provisioning/src/main/java/org/wildfly/build/util/POSIXFinePrivilegesHelper.java

```diff
@@ -0,0 +1,44 @@
+package org.wildfly.build.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+import static org.wildfly.build.provisioning.ServerProvisioner.OS_WINDOWS;
+
+public class POSIXFinePrivilegesHelper {
+
+    // TODO: ugly
+    private static final Path NO_PATH_PRODIVED = null;
+
+    private static final String JAR_FILE_MASK = "rw-r--r--";
+    private static final String PROPERTIES_FILE_MASK = "rw-------";
+
+    public static void fineTunePrivileges(Path path) throws IOException {
+        if (!OS_WINDOWS) Files.setPosixFilePermissions(path,getPermissionsMasks(NO_PATH_PRODIVED,path));
```

Comment:
Instead of just checking if it's Windows we should probably do something like: `FileSystems.getDefault().supportedFileAttributeViews().contains("posix")`